modules/objdetect/include/opencv2/objdetect/charuco_detector.hpp

```diff
@@ -16,6 +16,7 @@ struct CV_EXPORTS_W_SIMPLE CharucoParameters {
     CV_WRAP CharucoParameters() {
         minMarkers = 2;
         tryRefineMarkers = false;
+        checkBoardStructure = true;
```

Comment:
Why do you need extra flag?

Reply:
this flag allows you to return the old behavior or use your own board structure check.